### PR TITLE
Refactor taskAction resource to take in a Task definition uuid in add…

### PR DESCRIPTION
**Pull request by gitcliff**
link:https://issues.openmrs.org/browse/RESTWS-764

### CODE REVIEW

**Comment by gitcliff** [comment]
@ibacher kindly requesting for your review on this pr

File: omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskActionResource2_4.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+			break;
+		case SHUTDOWNTASK:
+			shutDownTasks(taskDefinitions);
+			break;
+		case RESCHEDULEALLTASKS:
+			reScheduleAllTasks();
+			break;
+		case RUNTASK:
+			runTasks(taskDefinitions);
+			break;
```
Comment (ibacher) [inline omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskActionResource2_4.java:91]: These `case` statements should be indented one more level.

File: omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskDefinitionResource2_4.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	public TaskDefinition save(TaskDefinition taskDefinition) {
+		taskServiceWrapper.saveTaskDefinition(taskDefinition);
+		return taskDefinition;
+	}
+
+	@Override
+	public TaskDefinition getByUniqueId(String uniqueId) {
+		TaskDefinition taskDefinition = taskServiceWrapper.getTaskByName(uniqueId);
+		if (taskDefinition == null) {
+			taskDefinition = taskServiceWrapper.getTaskByUuid(uniqueId);
```
Comment (ibacher) [inline omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskDefinitionResource2_4.java:118]: I can see why we want to *add* the functionality to get a task by UUID, but do we also want to remove the functionality to get the task by ID?

Comment (gitcliff) [reply omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskDefinitionResource2_4.java:118]: @ibacher yes i think its ideal to get a task by its UUID and leave the task by ID for internal database identification and operations

File: omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskActionResource2_4.java (hunk 1 above)
Comment (gitcliff) [reply omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskActionResource2_4.java:91]: yes sure

**Comment by gitcliff** [comment]
@ibacher thanks, sure i was going to add the tests .
This pr involves refactoring the `Task_Definition` resource because the `Task_Action` resource relies on its` getByUniqueId(String uniqueId)` of `Task_Definition` in order to do a POST  request of `Task_Action` with a UUID of a task definition just like the way the ` getTaskByName(name)` of task definition is called into action when doing a POST request of Task_Action with a task definition name..

File: omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskActionResource2_4.java (hunk 1 above)
Comment (gitcliff) [reply omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskActionResource2_4.java:91]: yes sure,going to add it

**Pull request closed without merging**